code/modules/projectiles/projectile.dm

```diff
@@ -1089,7 +1097,15 @@
 
 	while (isturf(loc) && !QDELETED(src))
 		process_movement(ICON_SIZE_ALL, hitscan = TRUE)
-		if (TICK_CHECK || paused || QDELETED(src))
+
+		if (QDELETED(src))
+			return
+
+		if (TICK_CHECK || paused)
+			create_hitscan_point()
+			// Create tracers if we get timestopped or lagchunk so there aren't weird delays
+			generate_hitscan_tracers(impact_point = FALSE, impact_visual = FALSE)
+			record_hitscan_start(offset = FALSE)
```

Comment:
how expensive is this?

Reply:
> how expensive is this?

~500us, tracers aren't costly by themselves but light that they can produce is. We could, theoretically, not generate tracer light if its created due to the projectile overtiming, or maybe instructing it to generate those next tick in free time.